src/components/ListV2/index.tsx

```diff
@@ -0,0 +1,139 @@
+import { ComponentProps, ReactNode } from 'react'
+import Stack from '../Stack'
+import { ListBody } from './ListBody'
+import { ListCell } from './ListCell'
+import { ListProvider } from './ListContext'
+import { ListExpandable } from './ListExpandable'
+import { ListHeader } from './ListHeader'
+import { ListHeaderRow } from './ListHeaderRow'
+import { ListHeaders } from './ListHeaders'
+import { ListLoadingPlaceholder } from './ListLoadingPlaceholder'
+import { ListRow } from './ListRow'
+
+type ListColumn = Omit<
+  ComponentProps<typeof ListHeader>,
+  'children' | 'colSpan'
+> & {
+  label: ReactNode
+  width?: string
+  id?: string
+}
+
+type ListProps<T> = {
+  children: ReactNode
+  /**
+   * Add checkboxes on the list
+   * */
+  selectable?: boolean
+  /**
+   * Row ids of the selected checkboxes
+   * */
+  selectedIds?: string[]
+  /**
+   * When selectedIds change
+   * */
+  onSelectedIdsChange?: (selectedIds: string[]) => void
+  data: T[]
+  /**
+   * The idKey of each data entry
+   * */
+  idKey?: string
+  /**
+   * Set it to true if you want to display a placeholder during loading
+   * */
+  isLoading?: boolean
+  /**
+   * Auto close opened expandable row when another is opening
+   * */
+  autoClose?: boolean
+
+  columns?: ListColumn[]
+  template?: string
+}
+
+export const List = <
+  T extends Record<string, unknown> = Record<string, unknown>,
+>({
+  children,
+  columns,
+  template,
+  selectable,
+  selectedIds,
+  onSelectedIdsChange,
+  data,
+  idKey,
+  isLoading,
+  autoClose,
```

Comment:
Adding `className` prop on `<Stack role="table"...>` might be interesting if we want to change gap or others properties